### ISSUE TITLE
[balance/ bugs] "Fatigue" vs "Infiltraition"

### ISSUE THREAD

**Issue by Crolock** [long-war-2/lwotc#859]
I am about half way through the campaign on veteran difficulty and had a pretty good time so far though there is this one thing I just need to discuss....

"Fatigue" vs "Infiltraition"

The thing that bothers me most about LWotC is the fact that instead of just having one "game mechanic" to limit the distribution of your forces (Fatigue in WotC, Infiltration in LW2) LWotC presents you with the worst of both worlds.
I particularly take issue with this as it encurages one specific playstyle while making others pretty much impossible to pull of. So what am I talking about?
Low infiltration missions!
They've been though already, but now they will just straight up f* your campaign if you would try to do more off those in a short time.  
In LW2 you could chain as many low infiltration missions as you wanted as long as you had the healthy soldiers to take part.
In WotC there where no high density missons and fatigue served mostly to mix up the squads a bit.
In LWotC however one low infiltration mission is likely to put your entire squad (or most of it anyway) into tired state. As will is reduced mostly by spotting enemies. So lets say 7 soldiers are out for 6 days on average! You only start with 30 (?) and you need to get them up to ssgt before they do much as well, not calculating in deaths you have to take in ironman playthroughs. So your entire roster is out for a week if you decide to do 2 low infiltration missions. 
Compare this to the optimal situation which takes 4 soldiers 4 days to infiltrate for an low density mission. 

So in the time of 16 days:  4x 4 Soldiers on 4 different low density, high infiltration missons take 4 days this comes to about 12-16 missions done in this timeframe (depending on mission spawns)
While in the time of 16 days: 2x 8 Soldiers on 2 different high density, low infiltration missons could only do 4-5 missions.

So why did we do those high density missions back in the day?
While there are certain objectives like full sweep troupe colums which profit from larger numbers of enemies, most of the time one would go for them to get the extra resources provided by that mission. Thus high risk: sending 8 good soldiers to face 32 aliens on high alert, for an arguably high reward like: get 1 extra scientist early. This was already a though decision to make, with Fatigue knocking your whole squad out on top its just a non factor.

Why is this an issue?

Well, as you could have guessed I liked the high  density missions, they are the reason I loved LW and LW2 fighting 4 vs 8 just feels so boring to me, but! there is an actual issue here:
The inconsistency of will restoration!

Bug? or intention?

A soldier leaving a mission in the READY (green) will state may immidiately go on another mission
and RECOVER WILL up to max DURING INFILTRATION.
However a soldier leaving a mission in the  TIRED (yellow) will state can also go on another mission but DOES NOT RECOVER WILL DURING INFILTRATION.
This makes high will soldiers which often desperatly lack other stats desirable simply because they might be able to 1 extra mission before tiring out. 
Now you might think will implants? No thanks! In the current state of the game a soldier tiers below 67% will so a 9 will implant adds only 3 will before tiring out, which is roughly 5-9 hostiles spotted. 

To make matters worse there are various questionable options for your tired soldiers to pass time.
I just don't think it is supposed to be like this:
- a tired soldier can not do covert ops (Approved)
- a tired soldier can train as officer and will be ready after training completes (I disapprove)
- two tired bondmates can lvl. their bond one of them will be ready afterwards (I disapprove)
- a tiered soldier can train psy (I disapprove)

Suggestion to fix this:
1. Tired soldiers like shaken soldiers shouldn't be able to do anything except go on another mission because they have to
2. Will should never recover during any activity (infiltration or other) to keep it consistent
3. Will recovery and Fatigue should be significantly reduced
I rewrote my ini to do the following and will probably update this after testing.

Tired threshold: 50%
Shaken threshold: 30%
Base Recovery: 0-3 days
Tired recovery: 2-4 days
Shaken recovery: 7-14 days

The fatigue system can help to prevent the player from doing one low infiltration mission after another. Soldiers should need to take a short brake after a while. with the base recovery of about 2 days and a threshold of 50 a soldier can either do 2 low or 1 high density mission without getting tiered, after which he should be left alone for 2 days to fully recover. No more "cheating" the system with recovery during long infiltrations, plus high density missions would not suffer so much from the added punishment.

Tldr:To me infiltration was ok, yet sometimes annoying to manage. I prefered the old LW2 fatigue. Slapping inconsistent WotC Fatigue onto the infiltration system though...just not fun!

**Comment by pledbrook** [comment]
Thanks for taking the effort to write your thoughts up. You make some good points and I agree the fatigue system could do with improving, although we may not agree exactly on the direction it should go.

First, I want to clear up some misconceptions:

* 4-soldier missions are risky, so your comparison should really be with the far more common 5-soldier missions.
* Tired soldiers do in fact recover Will during infiltration, it's just that you don't see the Will recovery until they hit 100%. So if they're Tired for 5 days and infiltrate for 6, they will be ready for action with 100% Will.

You also say:
> This makes high will soldiers which often desperatly lack other stats desirable simply because they might be able to 1 extra mission before tiring out.
Yes, and this is good. Why leave Will to be a completely useless stat?

> Now you might think will implants? No thanks! In the current state of the game a soldier tiers below 67% will so a 9 will implant adds only 3 will before tiring out, which is roughly 5-9 hostiles spotted.
And coping with an extra 5-9 hostiles is more than enough for the soldier not to tire on most missions.

It's also worth bearing in mind that 8-soldier squads will benefit from all that extra XP, so I don't see a problem with them suffering a bit more from fatigue. But I guess they're more likely to suffer from wounds.

All that said, we would rather have a system that reduces will per turn rather than per enemy spotted. It's just that the base game only implemented the latter, not the former.

I need to think about your suggestions. I'm not sure they are really what we want. The recovery from Tired seems a bit short. It really depends on what impact we want fatigue to have on normal infiltrated missions, underinfiltrated missions, and quick response missions.

**Comment by Crolock** [comment]
Thank you for take my thoughts into consideration.

> First, I want to clear up some misconceptions:
> 
>     4-soldier missions are risky, so your comparison should really be with the far more common 5-soldier missions.

I actually didn't know that I' ve been using only 4 soldiers on most low density missions so I'll keep that in mind.

> Tired soldiers do in fact recover Will during infiltration, it's just that you don't see the Will recovery >until they hit 100%. So if they're Tired for 5 days and infiltrate for 6, they will be ready for action with >100% Will.

Verify interesting indeed and something I did not notice during gameplay.
So if a soldier enters the mission marked as tired at an actual 90% will and ends that mission on 70% will would he be displayed as ready afterwards? (He should right?)

> It's also worth bearing in mind that 8-soldier squads will benefit from all that extra XP,
In fact I did not realize that either and I agree this is at least some redeeming quality. I initially thought mission xp would be split between soldiers.

As to my recent trial runs:
(tested about 4 ingame weeks)
Tired threshold: 50%
Shaken threshold: 30%
Base Recovery: 0-3 days 
Tired recovery: 2-4 days
Shaken recovery: 7-14 days

So this makes will pretty much a none factor. I will try again uping the numbers a bit.
Tired threshold: 50%
Shaken threshold: 30%
Base Recovery: 2-4 days 
Tired recovery: 4-6 days
Shaken recovery: 7-14 days

Another thought: 

> All that said, we would rather have a system that reduces will per turn rather than per enemy spotted. It's just that the base game only implemented the latter, not the former.

I also had this Idea. Though I feel like it would be difficult to balance this in a way so it doesn't just feel like a second timer as the timer in itself is already a huge point of controversy regarding x com 2.

Perhaps the implementation of will could be a way to simply replace the timer. 
Will would serve as a means to punish the player for taking to long, while there is no strict mission failed timer. The mission objective, whatever it is, still need to be full filled and continued waves of reinforcement could serve as an indirect timer, as they already do.

**Comment by gomegah** [comment]
Since we have a fatigue thread already, I would like to comment on that.

In my legend run, I had the feeling that the fatigue mechanic severly hampered my ability to get anything done during midgame; to a point where I almost had no progress in some months. Using the substitues' bench on retaliations did not work out too well for me on legend so I often had the following thing happen:
Advent retaliation --> Cancel an ongoing infiltration of a "good" squad; loose a few days.
When the retaliation is over, all soldiers will be tired for like 8-9 days, some will even be shaken. That means all soldiers will not be usable for another ~3 days; the shaken soldiers even a lot longer. With multiple retaliations per month it felt like half of my soldiers were permanently not available to actually progress through the campaign.
I lost the campaign in the late game, because I did not have enough facility leads and I think the soldiers-tired-because-of-retaliations problem played a major role in that.
Granted it was my first real try on a LW legend campaign and I certainly did not play optimally, I still feel the tired mechanic in its current state might be a bit too harsh.

**Comment by pledbrook** [comment]
Which version of the mod were you playing with? The Will loss mechanic is substantially different in dev build 13.

**Comment by gomegah** [comment]
I am not sure, I dled and installed beta 2 right after it was released. How can I find out, which dev build it is?

**Comment by harlequin565** [comment]
If you haven't downloaded anything since Beta 2 (like me) you're not on the latest version.

Just wanted to chime in with my Beta 2 experience (also Legend). I found that fatigue didn't hamper my run much at all. Because will recovers on infil, I didn't think anything of sending tired soldiers on missions if their recovery time was less than the infil time. There was a time when I had a string of retaliations within a 2 week period and towards the end, I was getting tired soldiers (which I sort of expected). The worst of it was the final 2 missions when I ended up with tired soldiers after the radio tower and no time to rest them before Leviathan. Note to self: Try to raise a team for each mission. Not looked at dev build 13 yet, so need to read that to discover changes.

**Comment by pledbrook** [comment]
Dev build 13 switched to the following system:

* All soldiers lose 1 Will point every 4 turns (that should probably be made configurable per difficulty)
* Soldiers can recover from 67% Will to max Will in 6 days, but take 8 days to recover from Tired (I suspect these numbers may need tuning)
* Soldiers no longer recover Will while infiltrating, training, being on haven duty, or on covert actions

Will is still an important stat because soldiers recover a *percentage* of their max Will each day, so they recover from missions more quickly the higher the max Will they have. It seems to work OK in my limited experience and makes you consider soldiers' current Will before deploying them. It's better to let them rest a few days while not Tired than let them get Tired on a mission.

Note that this build also makes the effects of Tired a lot less punishing as the chances for panics (including berserk, shattered and obsessed) have been significantly reduced.

**Comment by pledbrook** [comment]
Oh, I also wanted to say that the Will-loss-per-turn feature doesn't have much effect on most guerrilla ops, since the timers are pretty short. However, it does have an impact on untimed missions. Now you have to decide whether the time spent getting a perfect opener is worth the potential cost in lost Will. Especially as it could lead to soldiers becoming Tired even if they start the mission at max Will. This also means that high-Will soldiers should be favoured for long untimed missions.

**Comment by gomegah** [comment]
New will system makes a lot more sense to me.
Btw. thx @Pledbrook and the lwotc team for porting this awesome mod.  :)

**Comment by pledbrook** [comment]
I think most of the concerns in this issue are addressed by the above commit.
